torch/_dynamo/variables/nn_module.py

```diff
@@ -186,6 +190,45 @@ def record_nn_module_stack():
                 if is_lazy:
                     self.module_type = mod.cls_to_become
 
+                # Custom name denotes that we do not want to just call the module
+                if custom_name:
+                    # TODO(voz): Refactor this into a generic as_proxy() for nn module
+                    # We use variations of this pattern in a few places now.
+                    def make_attr(name):
+                        node = tx.output.create_proxy(
+                            "get_attr",
+                            name,
+                            tuple(),
+                            {},
+                        )
+                        return node
+
+                    # Bind in self
+                    name = self.module_key
+
+                    tx.output.register_attr_or_module(
+                        mod,
+                        name,
+                        name,
+                        source=NNModuleSource(GetItemSource(self.source, name)),
+                        **options,
+                    )
+                    proxy_for_mod = make_attr(name)
+                    proxy_for_mod.node.meta["example_value"] = mod
+
+                    proxy_args, proxy_kwargs = proxy_args_kwargs(args, kwargs)
+                    return variables.TensorVariable.create(
+                        tx=tx,
+                        proxy=tx.output.create_proxy(
+                            "call_method",
```

Comment:
It is not sound to insert an arbitrary method call into the graph.  Perhaps we should inline it instead with something like.
```
return UserMethodVariable(....).call_function(...)
```

Reply:
Why is it not sound?

Reply:
Couldn't the method contain arbitrary code which might not work with AotAutograd?

Reply:
Yes, but we would cover all those as either graph breaks, or with a downstream is_aot_autograd_safe_to_run right? I could totally be wrong here, but I figured mutation was the worst that could happen. Is there a test case you had in mind that I could add to exercise the arbitrary code getting missed here?

Reply:
Added a test that ensures `is_aot_autograd_safe_to_run` still reports correctly on a graph we don't otherwise break on.

Reply:
In your test case is `_some_bad_fwd()` put in the graph?  If so could it contain control flow based on data?

Reply:
> In your test case is _some_bad_fwd() put in the graph? 
No, not directly - the graph has its contents though:

```
def forward(self, arg0, arg1):
    _set_grad_enabled = torch._C._set_grad_enabled(False)
    add_ = arg0.add_(arg1);  arg0 = None
    _set_grad_enabled_1 = torch._C._set_grad_enabled(True)
    return arg1
 ```
 
 >  If so could it contain control flow based on data?
 
 Yes, but that would be a generic_jump, right? Added a test for that.

Reply:
Ah, my mistake.  I thought we could get a call to `_some_bad_fwd` in the graph.

Reply:
Fwiw, I thought this would do that too, and was planning on having to take the contents apart manually for it not to just be a blackbox of `call_method` into some unknown method that could contain anything. I tried this approach, and it already did the unpacking for us, which was nice. So its a very reasonable assumption, I could have done more in my PR to be clear about the behavior upfront.